cmd/charm/charmcmd/cmd_test.go

```diff
@@ -40,36 +45,45 @@ import (
 // its context directory set to dir. It returns the output of the command
 // and its exit code.
 func run(dir string, args ...string) (stdout, stderr string, exitCode int) {
-	return runWithInput(dir, "", args...)
-}
-
-// runWithInput runs a charm plugin subcommand with the given arguments,
-// its context directory set to dir, and with standard input set to in.
-// It returns the output of the command and its exit code.
-func runWithInput(dir string, in string, args ...string) (stdout, stderr string, exitCode int) {
 	// Remove the warning writer usually registered by cmd.Log.Start, so that
 	// it is possible to run multiple commands in the same test.
 	// We are not interested in possible errors here.
 	defer loggo.RemoveWriter("warning")
 	var stdoutBuf, stderrBuf bytes.Buffer
 	ctxt := &cmd.Context{
 		Dir:    dir,
-		Stdin:  strings.NewReader(in),
+		Stdin:  nil,
 		Stdout: &stdoutBuf,
 		Stderr: &stderrBuf,
 	}
 	exitCode = cmd.Main(charmcmd.New(), ctxt, args)
 	return stdoutBuf.String(), stderrBuf.String(), exitCode
 }
 
+type homeSuite struct{}
+
+func (s *homeSuite) Init(c *qt.C) {
+	oldHome := utils.Home()
+	c.AddCleanup(func() {
+		utils.SetHome(oldHome)
+	})
+	utils.SetHome(c.Mkdir())
+	c.Setenv("HOME", utils.Home())
+	err := os.MkdirAll(jujutesting.JujuXDGDataHomePath(), 0755)
+	c.Assert(err, qt.Equals, nil)
+	c.AddCleanup(func() {
+		osenv.SetJujuXDGDataHome("")
+	})
+	osenv.SetJujuXDGDataHome(jujutesting.JujuXDGDataHomePath())
+}
+
 // commonSuite sets up and tears down a charm store
```

Comment:
Similarly to homeSuite, it would be nice if this could be renamed to something not including the word "suite".
Maybe a similar pattern to what I suggested above might work here too? Maybe consider splitting it into domain-specific helpers too, if that makes sense.

I'm not sure whether embedding the common functionality makes as much sense now too - maybe put it in a named field. (that might be too disruptive, I guess, for this PR)